src/core/net/SRP_CODER.md

```diff
@@ -0,0 +1,349 @@
+
+## Remove Service Block
+
+### Remove Service Dispatch
+
+```
+    0      1     2     3     4     5     6     7
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+  |  0  |  1  |     (unused)                      |
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+```
+
+### Remove Service Block Format
+
+- “Remove Service Dispatch” byte
+- Service instance label
+- Service name labels (excludes zone name)
+
+## Host Block
+
+### Host Dispatch
+
+```
+    0      1     2     3     4     5     6     7
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+  |  1  |  0  | AT  | ADR | KT  | KEY | (unused)  |
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+```
+
+- `AT` flag (Address TTL) - `0` AAAA TTL elided (use default from header), `1`: TTL encoded.
+- `ADR` flag (Address list) - `0`: Address list elided (no addresses), `1`: Address list encoded.
+- 'KT flag (Key TTL) - `0`: Key TTL elided (use default from header), `1`: TTL encoded
+- `KEY` flag (Key) - `0`: Key elided, `1`: Key encoded.
+
+### Host Block Format
+
+- "Host Dispatch" byte
+- AAAA record TTL (if not elided)
+- Address list (if not empty based on `ADR` flag) - Starts with Address Dispatch byte
+- Key TTL (if not elided)
+- Key (if not elided) - 64 bytes
+
+TTL fields (when not elided) are encoded using the compact integer format.
+
+When the address list is not empty, each address begins with an Address Dispatch byte, followed by the specific information indicated by that dispatch byte.
+
+### Address Dispatch
+
+```
+    0      1     2     3     4     5     6     7
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+  |  C  |  M  | (unused)  |    Context ID         |
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+```
+
+- `C` flag (Context compression)
+  - `0`: IPv6 address fully encoded (16 bytes) after dispatch.
+  - `1`: Context compression used. Context ID provided in lower 4 bits. Address IID (8 bytes) encoded after dispatch.
+- `M` flag (More addresses)
+  - `0`: This is the last address in the list.
+  - `1`: There are more addresses in the list.
+
+When context compression is used, the corresponding prefix is determined using Thread Network Data and the specified Context ID, similar to 6LoWPAN IPv6 Header compression. Specifically, to determine the Context ID from Thread Network Data, the encoder MUST check that the "compress" bit in the corresponding 6LoWPAN ID TLV is set before using the Context ID and compressed format. The decoder MUST NOT check the "compress" bit when uncompressing the address.
+
+## Footer Block
+
+### Footer Dispatch
+
+```
+    0      1     2     3     4     5     6     7
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+  |  1  |  1  |  0  | LS  | KLS |     |   SIG     |
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+```
+
+- `LS` flag (Lease) - `0`: Lease elided (assume zero). `1`: Lease encoded
+- `KLS` flag (Key lease) - `0`: Key lease elided (assume zero). `1`: Key lease encoded
```

Comment:
When matter deletes a lease, it sets the key lease to 1209600. This is a big number, and might be a better default, since I think the key lease is never zero unless the accessory is going away forever, which I suspect is never known when an SRP update is sent.

Reply:
I should say that the key lease has the same value when the host lease is 7200, so pretty much any matter update is going to use that value.

Reply:
Good idea. We can change it to use the commonly default values.

Will update it in next push (when I get time to get back to this PR).

Reply:
Updated in new push, `7200` for lease, `1209600` for key lease